data-prepper-api/src/main/java/com/amazon/dataprepper/model/configuration/PipelineModel.java

```diff
@@ -12,12 +13,26 @@
  * @since 1.2
  */
 public class PipelineModel {
+    private static List<PluginModel> validateProcessor(final List<PluginModel> preppers, final List<PluginModel> processors) {
+        if (preppers != null && processors != null) {
+            String message = "Pipeline model cannot specify a prepper and processor configuration. It is " +
+                    "recommended to move prepper configurations to the processor section to maintain compatibility " +
+                    "with DataPrepper version 1.2 and above.";
+            throw new IllegalArgumentException(message);
+        }
+        else if (preppers != null) {
+            return preppers;
```

Comment:
It might be good to log a warning here noting that the `prepper:` configuration is being replaced by `processor:`.